bundles/test/org.eclipse.smarthome.test/src/main/java/org/eclipse/smarthome/test/java/JavaOSGiTest.java

```diff
@@ -53,19 +54,18 @@
 
     @Before
     public void bindBundleContext() {
-        bundleContext = initBundleContext();
+        bundleContext = getBundleContext();
         assertThat(bundleContext, is(notNullValue()));
     }
 
     /**
-     * Initialize the {@link BundleContext}, which is used for registration and unregistration of OSGi services.
-     *
-     * <p>
-     * This uses the bundle context of the test class itself.
+     * Returns the {@link BundleContext}, which is used for registration and unregistration of OSGi
+     * services. By default it uses the bundle context of the test class itself. This method can be overridden
+     * by concrete implementations to provide another bundle context.
      *
      * @return bundle context
      */
-    private @Nullable BundleContext initBundleContext() {
+    protected @Nullable BundleContext getBundleContext() {
```

Comment:
The private method `initBundleContext` is used to initialize the member variable `bundleContext`.

The member variable is protected, so it can be accessed by subclasses.
It is also annotated so we ignore the nullness annotations and the fact that it could be null.

If you want to access the bundleContext member by a function, please introduce a new one `getBundleContext()` that returns the member.

There is no need to use `FrameworkUtil` all the time to access the bundle context (it should be enough to do it once in `@Before`).